workshop/Lab3/README.md

```diff
@@ -0,0 +1,252 @@
+# Lab 3: Scale and update apps natively, building multi-tier applications.
+
+In this lab, understand how to build applications using basic building blocks which are Kubernetes objects such as pods, deployments and services. These objects are defined in configuration files using yaml. Once defined, these objects are created using Kubernetes CLI. In this lab we will explore how to build a multi-tier application that can scale.
+
+Before we work with the application we need to clone a github repo from which we can get all our configuration files.
+
+``` 
+	git clone git@github.com:IBM/guestbook.git
+```
+
+# 1. Scale apps natively
+
+Kubernetes accomplishes scaling out through a deployment.  By specifying a  replica number higher than 1 to run that many pods, you can ensure your deployment has the available resources to handle increasing load on your application.
+
+1. Consider the following deployment configuration for guestbook application
+    ```
+    apiVersion: apps/v1
+	kind: Deployment
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  replicas: 3
+	  selector:
+	    matchLabels:
+	      app: guestbook
+	  template:
+	    metadata:
+	      labels:
+	        app: guestbook
+	    spec:
+	      containers:
+	      - name: guestbook
+	        image: ibmcom/guestbook:v1
+	        ports:
+	        - name: http-server
+	          containerPort: 3000
+    ```
+The above configuration file create a deployment object named 'guestbook' with a pod containing a single container running the image 'ibmcom/guestbook:v1'.  Also the configuration specifies replicas set to 3 and kubernetes tries to make sure that at least three active pods running at any time.
+	
+- Create the pod with label app=myguestbook using pod.yaml 
+
+   ```kubectl create -f guestbook-deployment.yaml```
+
+- List the pod with label app=guestbook 
+
+   ```kubectl get pods -l app=guestbook```
+
+2. We now define a service object to expose the deployment to external clients.
+```
+	apiVersion: v1
+	kind: Service
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  ports:
+	  - port: 3000
+	    targetPort: http-server
+	  selector:
+	    app: guestbook
+	  type: LoadBalancer
+``` 
+The above configuration allows to create a service object named 'guestbook'  which will allow incoming trafiic for guestbook to port 3000 on one of the running pods. Port type 'LoadBalancer' is one of the types that exposes service using cloud providers loadbalancer to route trafiic to Nodeport or ClusterIP.
+
+- Let us now create guestbook service.
+		```kubectl create -f guestbook-service.yaml```
+		
+- Test guestbook app using a browser of your choice using the url '\<your-cluster-ip\>:\<node-port\>'
+
+You can simply change the deployment, alter replicas using the following command. Kubernetes then sees the desired the you want and tries to reconcile the actual current state to the desired stat. 
```

Comment:
This feels out of place, shouldn't this be under the Deployment section not the Service section?